src/python/pants/util/dirutil.py

```diff
@@ -314,6 +314,20 @@ def safe_rmtree(directory):
   else:
     shutil.rmtree(directory, ignore_errors=True)
 
+def safe_rmtree_recursive(directory):
```

Comment:
Personally, I prefer merging into `safe_rmtree` and introducing a kwarg `recursive=False`. It reduces code duplication and that's a sensible option for `safe_rmtree`.

Now that we are Py3 only, you can even enforce that the arg is kwarg only! 
```python
def safe_rmtree(directory: str, *, recursive: bool = False) -> None:
  ...
```

Reply:
i am not fussed either way - happy to make the change!

Reply:
Especially because we can enforce that it's kwarg only, definitely think it'd be good to make! (Unless others have a strong objection).

One important confirmation, though, this does not need to be cherry-picked into 1.17.x does it? If it does, then you can't use the `*` syntax.

Reply:
we might actually need to cherrypick it onto 1.17.x - if thats the case, I will leave them separate -- if not, will merge

Reply:
I'd still merge them and optimize for the long term solution here, since this is API public. There's nothing stopping using the kwarg in 1.17.x, we just couldn't enforce it like we could with 1.18.x.